edx_exams/apps/api/serializers.py

```diff
@@ -94,3 +95,56 @@ class Meta:
             "id", "created", "modified", "user", "start_time", "end_time",
             "status", "exam", "allowed_time_limit_mins", "attempt_number"
         )
+
+
+class StudentAttemptSerializer(serializers.ModelSerializer):
+    """
+    Serializer for the ExamAttempt model containing additional fields needed for the frontend UI
+    """
+
+    # directly from the ExamAttempt Model
+    attempt_id = serializers.IntegerField(source='id')
+    attempt_status = serializers.CharField(source='status')
+
+    # custom fields based on the ExamAttemptModel
+    course_id = serializers.SerializerMethodField()
+    exam_type = serializers.SerializerMethodField()
+    exam_display_name = serializers.SerializerMethodField()
+    exam_url_path = serializers.SerializerMethodField()
+    time_remaining_seconds = serializers.SerializerMethodField()
+
+    def get_course_id(self, obj):
+        return obj.exam.course_id
+
+    def get_exam_type(self, obj):
+        return obj.exam.exam_type
+
+    def get_exam_display_name(self, obj):
+        return obj.exam.exam_name
+
+    def get_exam_url_path(self, obj):
+        """
+        Getter for custom exam_url_path field
+        """
+        from edx_exams.apps.core.api import get_exam_url_path  # pylint: disable=import-outside-toplevel
```

Comment:
Putting the import here to avoid a cyclical import error (which is still happening). Should these be moved to a utils file instead?